roles/openshift_health_checker/openshift_checks/etcd_imagedata_size.py

```diff
@@ -0,0 +1,84 @@
+# pylint: disable=missing-docstring
+"""
+Ansible module for determining if the size of OpenShift image data exceeds a specified limit in an etcd cluster.
+"""
+
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException, get_var
+
+
+class EtcdImageDataSize(OpenShiftCheck):
+    """Check that total size of OpenShift image data does not exceed the recommended limit in an etcd cluster"""
+
+    name = "etcd_imagedata_size"
+    tags = ["etcd"]
+
+    def run(self, tmp, task_vars):
+        etcd_mountpath = self.get_etcd_mountpath(get_var(task_vars, "ansible_mounts"))
+        etcd_avail_diskspace = etcd_mountpath["size_available"]
+        etcd_total_diskspace = etcd_mountpath["size_total"]
+
+        etcd_imagedata_size_limit = get_var(task_vars,
+                                            "cli_etcd_max_image_data_size_bytes",
+                                            default=int(0.5 * float(etcd_total_diskspace - etcd_avail_diskspace)))
+
+        etcd_is_ssl = get_var(task_vars, "openshift", "master", "etcd_use_ssl", default=False)
+        etcd_port = get_var(task_vars, "openshift", "master", "etcd_port", default=2379)
+        etcd_hosts = get_var(task_vars, "openshift", "master", "etcd_hosts")
+
+        config_base = get_var(task_vars, "openshift", "common", "config_base")
+
+        cert = task_vars.get("etcd_client_cert", "".join([config_base, "/master/master.etcd-client.crt"]))
+        key = task_vars.get("etcd_client_key", "".join([config_base, "/master/master.etcd-client.key"]))
+        ca_cert = task_vars.get("etcd_client_ca_cert", "".join([config_base, "/master/master.etcd-ca.crt"]))
+
+        for etcd_host in list(etcd_hosts):
+            args = {
+                "size_limit_bytes": etcd_imagedata_size_limit,
+                "paths": ["/openshift.io/images", "/openshift.io/imagestreams"],
+                "host": etcd_host,
+                "port": etcd_port,
+                "protocol": "https" if etcd_is_ssl else "http",
+                "version_prefix": "/v2",
+                "allow_redirect": True,
+                "ca_cert": ca_cert,
+                "cert": {
+                    "cert": cert,
+                    "key": key,
+                },
+            }
+
+            etcdstats = self.module_executor("etcdstats", args, task_vars)
+
+            if etcdstats.get("rc", 0) > 0:
```

Comment:
> Could the rc be negative?

Sooo......

Technically RC is always 0-255. However today I saw Ansible reporting an RC of -11 for a real RC of 139. So no idea WTF is going on inside of Ansible somewhere but ... you may want to check that they're following spec before trusting that the field "rc" will be >= 0.